sql/src/main/java/io/crate/operation/merge/MergeOperation.java

```diff
@@ -21,83 +21,78 @@
 
 package io.crate.operation.merge;
 
-import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.base.Optional;
 import io.crate.breaker.RamAccountingContext;
-import io.crate.core.collections.Bucket;
-import io.crate.core.collections.Row;
 import io.crate.executor.transport.TransportActionProvider;
-import io.crate.operation.DownstreamOperation;
+import io.crate.metadata.Functions;
+import io.crate.metadata.ReferenceResolver;
 import io.crate.operation.ImplementationSymbolVisitor;
-import io.crate.operation.RowDownstreamHandle;
+import io.crate.operation.PageDownstream;
 import io.crate.operation.projectors.FlatProjectorChain;
 import io.crate.operation.projectors.ProjectionToProjectorVisitor;
+import io.crate.operation.projectors.ResultProvider;
+import io.crate.planner.RowGranularity;
 import io.crate.planner.node.dql.MergeNode;
 import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
 import org.elasticsearch.common.settings.Settings;
 
-import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.Executor;
 
 /**
  * merge rows - that's it
  */
-public class MergeOperation implements DownstreamOperation {
+@Singleton
+public class MergeOperation {
```

Comment:
please rename to PageDownstreamFactory or something like that.. this isn't a MergeOperation anymore.